src/decoder/mod.rs

```diff
@@ -334,8 +379,9 @@ impl<R: Read + Seek> Decoder<R> {
         self.width = self.get_tag_u32(ifd::Tag::ImageWidth)?;
         self.height = self.get_tag_u32(ifd::Tag::ImageLength)?;
         self.strip_decoder = None;
-        self.photometric_interpretation = match FromPrimitive::from_u32(
-            self.get_tag_u32(ifd::Tag::PhotometricInterpretation)?
+        // TODO: error on non-SHORT value.
```

Comment:
Shouldn't we just handle these errors right now?

Reply:
It's not straightforward since the `Value` does not store whether it was create from a `SHORT` (`u16`) or a `BYTE` (`u8`) or another compatible type. The specification only permits exactly the `SHORT` type so that this is strictly more permissive either. And for any conforming image the `as u16` cast will also not result in incorrectly interpreted values.

Since it would entail a fair bit of rewrite of the whole `Value` system I'd rather do this in a future PR. I'll open a tracking issue.

Reply:
Understandable